lib/quby/questionnaires/dsl/charting/chart_builder.rb

```diff
@@ -35,6 +35,30 @@ def y_categories(y_categories)
           @chart.y_categories = y_categories
         end
 
+        def y_range_categories(*y_range_categories)
+          if y_range_categories.length.even? || y_range_categories.length < 3
+            fail "#{@chart.key} y_range_categories should be of the form (0, 'label 0-10', 10, 'label 10-20', 20)"
+          end
+
+          # converts [0, 'label 0-10', 10, 'label 10-20', 20] into [0, 'label 0-10', 10, 10, 'label 10-20', 20]
+          expanded_categories =
+            y_range_categories[0..1] +
+            y_range_categories[2...-1].each_slice(2).flat_map { |(start_at, label)| [start_at, start_at, label] } +
+            y_range_categories[-1..-1]
+
+          # converts [0, 'label 0-10', 10, 10, 'label 10-20', 20] into
+          # {(0.0...10.0) => 'label 0-10', (10.0..20.0) => 'label 10-20'}
+          range_hash = {}
+          expanded_categories.each_slice(3).each_with_index.map do |(start_point, label, end_point), index|
+            if index < expanded_categories.length / 3 - 1
+              range_hash[start_point.to_f...end_point.to_f] = label
+            else # last range should be inclusive its end point
+              range_hash[start_point.to_f..end_point.to_f] = label
+            end
+          end
+          @chart.y_range_categories = range_hash
```

Comment:
Really? Je ziet geen voordeel in kleine functies, kleine classes, code die te begrijpen zonder eerst de comments te lezen en daarna nog een keer langzaam door de code te stappen?
Een nieuwe functie van 16 regels code met de complexiteit van een 30-regelige functie toevoegen zie je niet als probleem?